.github/workflows/benchmark.yml

```
@@ -0,0 +1,68 @@
+name: Run benchmarks
+
+on:
+  push:
+    branches: [setup_action]
+  pull_request:
+    types: [closed]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    defaults:
+      run:
+        shell: bash -l {0}
+    steps:
+      - name: Checkout repo content
+        uses: actions/checkout@v2
+
+      - name: Cache conda
+        uses: actions/cache@v2
+        env:
+          CACHE_NUMBER: 0
+        with:
+          path: ~/conda_pkgs_dir
+          key: ${{ runner.os }}-conda-${{ env.CACHE_NUMBER }}-${{
+            hashFiles('environment.yml') }}
+
+      - name: Setup miniconda
+        uses: conda-incubator/setup-miniconda@v2
+        with:
+          python-version: 3.8
+          activate-environment: pdist_aggregation
+          environment-file: environment.yml
+          channels: conda-forge
+          channel-priority: flexible
+          use-only-tar-bz2: true
+
+      - name: Set config vars for parallelism
+        run: |
+          export NUM_CORES=`python -c "import joblib; print(joblib.cpu_count())"`
+          export MKL_NUM_THREADS=$NUM_CORES
+          export OMP_NUM_THREADS=$NUM_CORES
+          export OPENBLAS_NUM_THREADS=$NUM_CORES
```

We need to avoid threads over-subscription when using BLAS/MKL within code parallelised by OpenMP, which must happen here.

I think of adding this in another PR improving benchmarks which would introduce `threadpoolctl` to do so. For the time being we can solve it setting the environment variables accordingly.
```suggestion
          export OMP_NUM_THREADS=$NUM_CORES
          # Avoid threads over-subscription in parallel implementation 
          # TODO: remove once threadpoolctl is set.
          export MKL_NUM_THREADS=1
          export OPENBLAS_NUM_THREADS=1
```